tests/beman/inplace_vector/spec.test.cpp

```diff
@@ -0,0 +1,716 @@
+    X c(n);
+    EXPECT_TRUE((std::is_same_v<decltype(c.empty()), bool>));
+    EXPECT_EQ(c.empty(), c.begin() == c.end());
+  }
+  // How to test complexity?
+}
+
+// Still [container.reqmts] (23.2.2.2):
+// Unless otherwise specified (see [associative.reqmts.except],
+// [unord.req.except], [deque.modifiers], [inplace.vector.modifiers], and
+// [vector.modifiers]) all container types defined in this Clause meet the
+// following additional requirements:
+// - If an exception is thrown by an insert() or emplace() function while
+//   inserting a single element, that function has no effects.
+//   --> specified in [inplace.vector.modifiers]
+// - If an exception is thrown by a push_back(), push_front(), emplace_back(),
+//   or emplace_front() function, that function has no effects.
+//   --> push_front()/emplace_front() n.a. for inplace_vector,
+//       push_back()/emplace_back() specified in [inplace.vector.modifiers]
+// - No erase(), clear(), pop_back() or pop_front() function throws an
+//   exception.
+//   --> erase() specified in [inplace.vector.modifiers], pop_front()
+//       n.a. for inplace_vector
+TEST(ContainerRequirements, NothrowClear) {
+  using T = int;
+  using X = inplace_vector<T, 5>;
+
+  EXPECT_TRUE(noexcept(std::declval<X>().clear()));
+}
+TEST(ContainerRequirements, NothrowPopBack) {
+  using T = int;
+  using X = inplace_vector<T, 5>;
+
+  X::size_type const values[]{0, X::max_size() > 0 ? 1 : 0, X::max_size() / 2,
+                              X::max_size() > 0 ? X::max_size() - 1 : 0,
+                              X::max_size()};
+  // pop_back() has a narrow contract, therefore we cannot check noexcept().
+  for (X::size_type n : values) {
+    X c(n);
+    if (n > 0) {
+      EXPECT_NO_THROW(c.pop_back());
+    }
+  }
+}
+// - No copy constructor or assignment operator of a returned iterator throws an
+//   exception.
+TEST(ContainerRequirements, NothrowIterator) {
+  using T = int;
+  using X = inplace_vector<T, 5>;
+
+  EXPECT_TRUE(std::is_nothrow_copy_constructible_v<X::iterator>);
+  EXPECT_TRUE(std::is_nothrow_copy_constructible_v<X::const_iterator>);
+  EXPECT_TRUE(std::is_nothrow_move_constructible_v<X::iterator>);
+  EXPECT_TRUE(std::is_nothrow_move_constructible_v<X::const_iterator>);
+  EXPECT_TRUE(std::is_nothrow_copy_assignable_v<X::iterator>);
+  EXPECT_TRUE(std::is_nothrow_copy_assignable_v<X::const_iterator>);
+  EXPECT_TRUE(std::is_nothrow_move_assignable_v<X::iterator>);
+  EXPECT_TRUE(std::is_nothrow_move_assignable_v<X::const_iterator>);
+}
+// - No swap() function throws an exception.
+//   --> Specified in [inplace.vector.overview]
+// - No swap() function invalidates any references, pointers, or iterators
+//   referring to the elements of the containers being swapped.
+//   --> Waived by previous paragraph in [container.reqmts]
+
+// 3 For any N, inplace_vector<T, N>::iterator and inplace_vector<T,
+// N>::const_iterator meet the constexpr iterator requirements.
+//
+// TODO: Test this.
```

Comment:
There are variations of constexpr requirement, and this depends on to be deprecated trivial type trait. We should split this out into individual files.

Specifically:
- <= C++26 constexpr when is_trivial<T>
- >C++26 constexpr for wider T using trivial union, omit any reference to is_trivial<T>

Reply:
The mentioned requirement refers to the _iterators_ of inplace_vector, not inplace_vector itself. The constexpr requirement for the iterators is independent from T and its triviality, there is no change planned in >C++26, afaik.

Reply:
Ah I commented on the wrong lines, should be:

https://github.com/bemanproject/inplace_vector/blob/06855a2d83eca8a4054730cf0737b0eb593217e3/tests/beman/inplace_vector/spec.test.cpp#L518-L521